presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -56,8 +57,16 @@
     public OrcReader(OrcDataSource orcDataSource, MetadataReader metadataReader)
             throws IOException
     {
+        this(orcDataSource, metadataReader, new AtomicLong());
+    }
+
+    // This is based on the Apache Hive ORC code
+    public OrcReader(OrcDataSource orcDataSource, MetadataReader metadataReader, AtomicLong deltaMemory)
```

Comment:
I don't like passing an `AtomicLong` around. It doesn't need to be atomic. Create an object for this with a clearly defined API.